framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/AzureQueueStorageOutputChannel.cs

```diff
@@ -0,0 +1,263 @@
+            : base(factory)
+        {
+            //To be implemented
+        }
+
+        #region IOutputChannel_Properties
+        EndpointAddress IOutputChannel.RemoteAddress
+        {
+            get
+            {
+                return this.remoteAddress;
+            }
+        }
+
+        Uri IOutputChannel.Via
+        {
+            get
+            {
+                return this.via;
+            }
+        }
+        #endregion
+
+        public override T GetProperty<T>()
+        {
+            if (typeof(T) == typeof(IOutputChannel))
+            {
+                return (T)(object)this;
+            }
+
+            T messageEncoderProperty = this.encoder.GetProperty<T>();
+            if (messageEncoderProperty != null)
+            {
+                return messageEncoderProperty;
+            }
+
+            return base.GetProperty<T>();
+        }
+
+        /// <summary>
+        /// Open the channel for use. We do not have any blocking work to perform so this is a no-op
+        /// </summary>
+        protected override void OnOpen(TimeSpan timeout)
+        {
+        }
+
+        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            return new CompletedAsyncResult(callback, state);
+        }
+
+        protected override void OnEndOpen(IAsyncResult result)
+        {
+            CompletedAsyncResult.End(result);
+        }
+
+
+        #region Socket_Shutdown
+        /// <summary>
+        /// Shutdown ungracefully
+        /// </summary>
+        protected override void OnAbort()
+        {
+
+        }
+
+        /// <summary>
+        /// Shutdown gracefully
+        /// </summary>
+        protected override void OnClose(TimeSpan timeout)
+        {
+
+        }
+
+        protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            this.OnClose(timeout);
+            return new CompletedAsyncResult(callback, state);
+        }
+
+        protected override void OnEndClose(IAsyncResult result)
+        {
+            CompletedAsyncResult.End(result);
+        }
+        #endregion
+
+        #region Send_Synchronous
+        /// <summary>
+        /// Address the Message and serialize it into a byte array.
+        /// </summary>
+        ArraySegment<byte> EncodeMessage(Message message)
+        {
+            try
+            {
+                this.remoteAddress.ApplyTo(message);
+                return encoder.WriteMessage(message, int.MaxValue, parent.BufferManager);
+            }
+            finally
+            {
+                // We have consumed the message by serializing it, so clean up
+                message.Close();
+            }
+        }
+
+        public void Send(Message message)
+        {
+            try
+            {
+                queueClient.SendMessage(message.ToString());
+            }
+            catch
+            {
+
+            }
```

Comment:
You should wrap the exception in a CommunicationException and throw. Is there a specific exception type that will always be thrown? If so, is there any useful information you can get to use as part of the exception message? If not, just use the caught exception message as the message for the CommunicationException so they don't need to drill into the inner exception to get that info.

Reply:
makes sense. I wasnt sure what exception to be caught, so i had kept it empty to be filled up later but CommunicationException  does make sense.